pkg/frontend/admin_openshiftcluster_kubernetespods_logs_test.go

```diff
@@ -0,0 +1,212 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+	mock_adminactions "github.com/Azure/ARO-RP/pkg/util/mocks/adminactions"
+)
+
+func TestAdminKubernetesGetPodLogs(t *testing.T) {
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+	mockTenantID := "00000000-0000-0000-0000-000000000000"
+	method := http.MethodGet
+	ctx := context.Background()
+
+	type test struct {
+		name           string
+		namespace      string
+		podName        string
+		containerName  string
+		mocks          func(*test, *mock_adminactions.MockKubeActions)
+		wantStatusCode int
+		wantResponse   []byte
+		wantError      string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:          "cluster exist in db",
+			namespace:     "openshift-project",
+			podName:       "config",
+			containerName: "container-1",
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGetPodLogs(gomock.Any(), tt.namespace, tt.podName, tt.containerName).
+					Return([]byte(`fake logs`), nil)
+
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse:   []byte(`fake logs` + "\n"),
+		},
+		{
+			name:           "customer namespace, validation should fail",
+			namespace:      "customer-namespace",
+			podName:        "config",
+			containerName:  "container-1",
+			mocks:          func(tt *test, k *mock_adminactions.MockKubeActions) {},
+			wantStatusCode: http.StatusForbidden,
+			wantError:      "403: Forbidden: : Access to the provided namespace 'customer-namespace' is forbidden.",
+			//wantResponse:   []byte(`fake logs` + "\n"),
+		},
+		{
+			name:           "empty container name",
+			namespace:      "openshift-namespace",
+			podName:        "config",
+			containerName:  "",
+			mocks:          func(tt *test, k *mock_adminactions.MockKubeActions) {},
+			wantStatusCode: http.StatusBadRequest,
+			wantError:      "400: InvalidParameter: : The provided container name '' is invalid.",
+			//wantResponse:   []byte(`fake logs` + "\n"),
+		},
+	} {
+		t.Run(fmt.Sprintf("%s: %s", method, tt.name), func(t *testing.T) {
+			resourceID := fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID)
+			ti := newTestInfra(t).WithOpenShiftClusters().WithSubscriptions()
+			defer ti.done()
+			k := mock_adminactions.NewMockKubeActions(ti.controller)
+
+			tt.mocks(tt, k)
```

Comment:
Tip: you can do this:

```suggestion
			if tt.mocks != nil {
				tt.mocks(tt, k)
			}
```

So you don't have to add `mocks:          func(tt *test, k *mock_adminactions.MockKubeActions) {},` in every test case.

Reply:
okay ..

Reply:
done